### PR TITLE
[ISSUE #660] Add namespace in java client

### LINKED ISSUE #660: [Enhancement] Add namespace in java client
### Before Creating the Enhancement Request

- [X] I have confirmed that this should be classified as an enhancement rather than a bug/feature.

### Programming Language of the Client

Java

### Summary

Add namespace in java client.

### Motivation

For the purpose for the future namespace feature.

### Describe the Solution You'd Like

Add namespace in ClientConfiguration.

### Describe Alternatives You've Considered

No

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by drpmma**
### Which Issue(s) This PR Fixes

Fixes #660

### Brief Description

Add namespace in java client

### CODE REVIEW

File: java/client-apis/pom.xml [hunk 1]
```diff
@@ -34,5 +34,9 @@
             <groupId>com.google.guava</groupId>
             <artifactId>guava</artifactId>
         </dependency>
+        <dependency>
+            <groupId>org.apache.commons</groupId>
+            <artifactId>commons-lang3</artifactId>
```
Comment (aaron-ai) [inline java/client-apis/pom.xml:39]: Is it necessary to introduce the new dependence here? The shaded jar is fat enough...

File: java/client/src/main/java/org/apache/rocketmq/client/java/impl/consumer/ConsumerImpl.java [hunk 2]
```diff
@@ -123,7 +123,10 @@ protected ListenableFuture<ReceiveMessageResult> receiveMessage(ReceiveMessageRe
     }
 
     private AckMessageRequest wrapAckMessageRequest(MessageViewImpl messageView) {
-        final Resource topicResource = Resource.newBuilder().setName(messageView.getTopic()).build();
+        final Resource topicResource = Resource.newBuilder()
+            .setResourceNamespace(clientConfiguration.getNamespace())
```
Comment (aaron-ai) [inline java/client/src/main/java/org/apache/rocketmq/client/java/impl/consumer/ConsumerImpl.java:127]: Does `namespace` make sense for consumer group?

Comment (drpmma) [reply java/client/src/main/java/org/apache/rocketmq/client/java/impl/consumer/ConsumerImpl.java:127]: Yes, for sure.

Comment (drpmma) [reply java/client/src/main/java/org/apache/rocketmq/client/java/impl/consumer/ConsumerImpl.java:127]: Group's namespace is added in getProtobufGroup() method.

File: java/client-apis/pom.xml (hunk 1 above)
Comment (drpmma) [reply java/client-apis/pom.xml]: fixed

**Review by aaron-ai — APPROVED** [review]

**Pull request merged**